src/main/java/org/apache/mesos/state/CuratorStateStore.java

```diff
@@ -13,97 +13,107 @@
 import org.slf4j.LoggerFactory;
 
 /**
- * A CuratorStateStore stores TaskInfo and TaskStatus data in Zookeeper.
+ * CuratorStateStore is an implementation of {@link StateStore} which persists data in Zookeeper.
  *
  * The ZNode structure in Zookeeper is as follows:
- * rootPath
- *     -> ExecutorID-0
- *         -> TaskName-0
- *             -> TaskInfo
- *             -> TaskStatus
- *         -> TaskName-1
- *             -> TaskInfo
- *             -> TaskStatus
+ * <code>
+ * rootPath/
+ *     -> "FrameworkID"
+ *     -> ExecutorID-0/
+ *         -> TaskName-0/
+ *             -> "TaskInfo"
+ *             -> "TaskStatus"
+ *         -> TaskName-1/
+ *             -> "TaskInfo"
+ *             -> "TaskStatus"
  *         -> ...
- *    -> ExecutorID-1
- *         -> TaskName-0
- *             -> TaskInfo
- *             -> TaskStatus
- *         -> TaskName-1
- *             -> TaskInfo
- *             -> TaskStatus
+ *    -> ExecutorID-1/
+ *         -> TaskName-0/
+ *             -> "TaskInfo"
+ *             -> "TaskStatus"
+ *         -> TaskName-1/
+ *             -> "TaskInfo"
+ *             -> "TaskStatus"
  *         -> ...
  *    -> ...
+ * </code>
  */
-public class CuratorStateStore extends CuratorPersister implements StateStore {
+public class CuratorStateStore implements StateStore {
+
+    private static final Logger logger = LoggerFactory.getLogger(CuratorStateStore.class);
+
     private static final String TASK_INFO_PATH_NAME = "TaskInfo";
     private static final String TASK_STATUS_PATH_NAME = "TaskStatus";
     private static final String FWK_ID_PATH_NAME = "FrameworkID";
 
-    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final CuratorPersister curator;
     private final String rootPath;
     private final String fwkIdPath;
 
     public CuratorStateStore(String rootPath, String connectionString, RetryPolicy retryPolicy) {
-        super(connectionString, retryPolicy);
+        this.curator = new CuratorPersister(connectionString, retryPolicy);
         this.rootPath = rootPath;
         this.fwkIdPath = rootPath + "/" + FWK_ID_PATH_NAME;
     }
 
     @Override
     public void storeFrameworkId(Protos.FrameworkID fwkId) throws StateStoreException {
         try {
-            logger.info(String.format("Storing FrameworkID in '%s'", fwkIdPath));
-            store(fwkIdPath, fwkId.toByteArray());
+            logger.info("Storing FrameworkID in '{}'", fwkIdPath);
```

Comment:
@nickbp:  I chose the wrong log level for these kinds of statements.  They should all be changed to debug in my opinion.